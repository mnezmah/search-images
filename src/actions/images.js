import axios from 'axios'

export const  GET_IMAGES = 'GET_IMAGES'

function setImages(images) {
  return {
    type: GET_IMAGES,
    payload: images
  }
}

export function getImages(query) {
  return function(dispatch) {
    const apiKey = '14512980-f807f7da94eecb11fe86f08e9'
    axios.get(`https://pixabay.com/api/?key=${apiKey}&q=${query}&image_type=photo`)
      .then((response) => {
        console.log('response : ', response)
        console.log('data:', response.data)
        dispatch(setImages(response.data.hits))
      }).catch(error => console.log(error))
  }
}