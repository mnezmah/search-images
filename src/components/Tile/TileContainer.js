import React from 'react'
import { connect } from 'react-redux'
import Tile from './Tile'


 function TileContainer(props) {
   console.log('props: ', props)
  return (
    <div>
     { props.images.length ? 
      <Tile key = {props.images[0].id}
      imageUrl= {props.images[0].webformatURL}
      imageLabel={props.images[0].tags}
      /> : 'loading...'}
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    images: state.images
  }
}

export default connect(mapStateToProps)(TileContainer)