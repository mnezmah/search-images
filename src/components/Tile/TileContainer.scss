@import "../../Styling/main";

.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  position: relative;

  .image-container {
    width: 18rem;
    height: 12rem;

    @media (min-width: 480px) and (max-width: 599.5px) {
      width: 13rem;
      height: 12rem;
    }

    @media (min-width: 600px) and (max-width: 617.5px) {
      width: 15rem;
      height: 12rem;
    }

    @media (min-width: 618px) and (max-width: 739.5px) {
      width: 16rem;
      height: 12rem;
    }

    @media (min-width: 740px) and (max-width: 829.5px) {
      width: 19rem;
      height: 13rem;
    }

    @media (min-width: 830px) and (max-width: 889.5px) {
      width: 22rem;
      height: 15rem;
    }

    @media (min-width: 890px) and (max-width: 959.5px) {
      width: 15rem;
      height: 11rem;
    }

    @media (min-width: 960px) and (max-width: 1023.5px) {
      width: 16rem;
      height: 12rem;
    }

    @media (min-width: 1024px) and (max-width: 1294.5px) {
      width: 18rem;
      height: 12rem;
    }

    @media (min-width: 1295px) and (max-width: 1599.5px) {
      width: 22rem;
      height: 16rem;
    }
    @media (min-width: 1600px) and (max-width: 1911.5px) {
      width: 26rem;
      height: 18rem;
    }

    @media (min-width: 1912px) and (max-width: 2199.5px) {
      width: 30rem;
      height: 21rem;
    }

    @media (min-width: 2200px) {
      width: 26vw;
      height: 33vh;
    }
  }

  .title {
    display: inline-block;
    color: $main-white;
    max-width: 77vw;
    text-align: center;

    @media #{$screen-mobile-plus-only} {
      font-size: 0.9rem;
      max-width: 47vw;
    }

    @media #{$screen-tablet-portrait-only} {
      font-size: 16px;
      max-width: 12rem;
    }

    @media #{$screen-tablet-landscape-only} {
      font-size: 1rem;
      max-width: 25vw;
    }

    @media #{$screen-laptop-and-bigger} {
      font-size: 1.3rem;
      max-width: 17rem;
    }
  }
}

.message {
  margin: 0 auto;
  color: $main-white;

  @media #{$screen-tablet-portrait-and-bigger} {
    font-size: 1.5rem;
  }
}
