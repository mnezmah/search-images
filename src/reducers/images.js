import { GET_IMAGES } from '../actions/images'

const initialState = []

export default (state = initialState, action = []) => {
  switch (action.type) {
    case GET_IMAGES:
      return action.payload
      default: return state
  } 
}