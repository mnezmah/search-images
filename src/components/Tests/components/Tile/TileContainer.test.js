import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import TileContainer from '../../../Tile/TileContainer'
import images from '../../fixtures/images'

const mockStore = configureMockStore();
const store = mockStore({});

test('should render TileContainer corectly', () => {
  const wrapper = shallow(
    <Provider store={store}>
      <TileContainer />
    </Provider>
  )
  expect(wrapper).toMatchSnapshot()
})

test('should render Tiles corectly', () => {
  const wrapper = shallow(
    <Provider store={store}>
      <TileContainer images={images} />
    </Provider>
  )
  expect(wrapper).toMatchSnapshot()
})