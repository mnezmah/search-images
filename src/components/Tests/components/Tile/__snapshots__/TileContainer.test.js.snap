// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TileContainer corectly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(TileContainer) />
</ContextProvider>
`;

exports[`should render Tiles corectly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(TileContainer)
    images={
      Array [
        Object {
          "id": 1,
          "tags": "image1",
          "webformatURL": "hhtps://www.image1",
        },
        Object {
          "id": 2,
          "tags": "image2",
          "webformatURL": "hhtps://www.image2",
        },
        Object {
          "id": 3,
          "tags": "image3",
          "webformatURL": "hhtps://www.image3",
        },
      ]
    }
  />
</ContextProvider>
`;
